packages/openactive-broker-microservice/app.js

```diff
@@ -62,6 +71,108 @@ const startTime = new Date();
 let datasetSiteJson = {
 };
 
+/**
+ * Use OpenActive validator to validate the RPDE item
+ *
+ * @param {any} body the data item
+ * @param {boolean} isOrdersFeed Whether this is an orders feed
+ */
+async function validateItem(body, isOrdersFeed) {
+  /**
+   * @type {{
+   *   loadRemoteJson: boolean,
+   *   remoteJsonCachePath: string,
+   *   remoteJsonCacheTimeToLive: number,
+   *   validationMode?: string,
+   * }} & typeof options
+   */
+  const optionsWithRemoteJson = {
+    loadRemoteJson: true,
+    remoteJsonCachePath: '../openactive-integration-tests/tmp',
+    remoteJsonCacheTimeToLive: 3600,
+    validationMode: isOrdersFeed ? 'OrdersFeed' : 'BookableRPDEFeed',
+  };
+
+  const errors = (await validate(body, optionsWithRemoteJson))
+    .filter((result) => result.severity === 'failure');
+
+  return errors;
+}
+
+const validationResults = new Map();
+
+/**
+ * Use OpenActive validator to validate the opportunity
+ *
+ * @param {any} data opportunity JSON-LD object
+ */
+async function storeValidationResults(data) {
```

Comment:
This function both validates and stores results, so maybe should be called something like `validateAndStoreValidationResults`?

Reply:
Good shout, have updated